docker_images/sklearn/tests/test_api_tabular_classification.py

```diff
@@ -67,26 +76,90 @@ def test_success_code(self):
-            response = client.post("/", data=b"Where do I live ?")
+            response = client.post("/", json={"inputs": data})
 
-        self.assertEqual(
-            response.status_code,
-            400,
-        )
+        assert response.status_code == 400
         content = json.loads(response.content)
-        self.assertEqual(set(content.keys()), {"error"})
+        assert "error" in content
+        assert "warnings" in content
+        wrong_versions = [
+            w for w in content["warnings"] if "Trying to unpickle estimator" in w
+        ]
+        assert len(wrong_versions)
+        error_message = json.loads(content["error"])
+        assert error_message["output"] == self.expected_output
+
+    def test_cannot_load_model(self):
+        # test the error message when the model cannot be loaded on a wrong
+        # sklearn version
+        self._check_requirement(not self.case_data["loads_on_new_sklearn"])
 
-    def test_missing_columns(self):
-        data = self.data["data"].copy()
-        data.pop(next(iter(data.keys())))
+        data = self.data
+        with TestClient(self.app) as client:
+            response = client.post("/", json={"inputs": data})
+
+        assert response.status_code == 400
+        content = json.loads(response.content)
+        assert "error" in content
+        assert "An error occurred while loading the model:" in content["error"]
+
+    @parameterized.expand(
+        [
+            ("add", "The following columns were given but not expected:"),
+            ("drop", "The following columns were expected but not given:"),
+        ]
+    )
+    def test_extra_columns(self, column_operation, warn_message):
+        # Test that the right warning is raised when there are extra columns in
+        # the input.
+        self._check_requirement(self.case_data["has_config"] and self._can_load())
+
+        data = self.data.copy()
+        if column_operation == "drop":
+            # we remove the first column in the data. Note that `data` is a
+            # dict of column names to values.
+            data["data"].pop(next(iter(data["data"].keys())))
+        elif column_operation == "add":
+            # we add an extra column to the data, the same as the first column.
+            # Note that `data` is a dict of column names to values.
+            data["data"]["extra_column"] = next(iter(data["data"].values()))
 
-        inputs = {"data": data}
         with TestClient(self.app) as client:
-            response = client.post("/", json={"inputs": inputs})
+            response = client.post("/", json={"inputs": data})
 
-        self.assertEqual(
-            response.status_code,
-            400,
-        )
+        assert response.status_code == 400
+        content = json.loads(response.content)
+        assert "error" in content
+        assert "warnings" in content
+
+        wrong_versions = [w for w in content["warnings"] if warn_message in w]
+        assert len(wrong_versions)
```

Comment:
Again, those two lines could be simplified to

`assert any(warn_message in w for w in content["warnings"])`

If not: I think the variable name `wrong_versions` is misleading, since we deal with wrong columns, not wrong versions, right?